avocado/utils/network/interfaces.py

```diff
@@ -56,8 +56,12 @@ def _get_interface_details(self, version=4):
         cmd = "ip -{} -j address show {}".format(version, self.name)
         output = run_command(cmd, self.host)
         try:
-            return json.loads(output)
-        except json.JSONDecodeError:
+            result = json.loads(output)
+            for item in result:
+                if item.get('ifname') == self.name:
+                    return item
+            raise NWException("Interface not found")
```

Comment:
This exception will be caught by the next line here, is it intended?

Reply:
yes. it was. Just to avoid duplicate lines to log.